lib/msf/core/post/file.rb

```diff
@@ -607,6 +613,34 @@ def copy_file(src_file, dst_file)
 
 protected
 
+
+  def _write_file_powershell(file_name, data)
+    offset = 0
+    chunk_size = 16256
+    loop do
+      _write_file_powershell_fragment(file_name, data, offset, chunk_size)
+      offset += chunk_size + 1
+      break if offset >= data.length
+    end
+  end
+
+  def _write_file_powershell_fragment(file_name, data, offset, chunk_size)
+  	chunk = data[offset..offset+chunk_size]
+  	length = chunk.length
+  	compressed_chunk = Rex::Text.gzip(chunk)
+  	encoded_chunk = Base64.strict_encode64(compressed_chunk)
+    pwsh_code = %($encoded=\"#{encoded_chunk}\";
+    $mstream = [System.IO.MemoryStream]::new([System.Convert]::FromBase64String($encoded));
+    $reader = [System.IO.StreamReader]::new([System.IO.Compression.GZipStream]::new($mstream,[System.IO.Compression.CompressionMode]::Decompress));
+    $filename = [System.IO.File]::Open('#{file_name}', [System.IO.FileMode]::#{offset == 0 ? "Create" : "Append"})
+    $file_bytes=[System.Byte[]]::CreateInstance([System.Byte],#{length});
+    $reader.BaseStream.Read($file_bytes,0,$file_bytes.Length);
+    $filename.Write($file_bytes, 0, $file_bytes.Length);
+    $filename.Close();
+    $mstream.Close();
+    $reader.Close();)
+    cmd_exec(pwsh_code)
+
```

Comment:
There's a missing `end` here.  That's why the rspec tests are failing:
![image](https://user-images.githubusercontent.com/17987018/131541370-5c27ba60-3eb0-4ec1-a040-f4843f0ffc68.png)

Reply:
I'm not entirely sure why the Linux sanity tests are failing.   I'm thinking it may just be a failure in the test system.